### PR TITLE
Label PRs

### PR DESCRIPTION
**Pull request by jhaals**
Adds automatic labelling of PRs based on the paths edited.
Starting with scaffolder, catalog and search.
This is primary to improve visibility and filtering for interested parties

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+catalog:
+  - plugins/catalog/**/*
+  - plugins/catalog-*/**/*
```
Comment (benjdlambert) [inline .github/labeler.yml:3]: `packages/catalog-*`

Comment (jhaals) [reply .github/labeler.yml:3]: good point!

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  - packages/catalog-*/**/*
+scaffolder:
+  - plugins/scaffolder/**/*
+  - plugins/scaffolder-*/**/*
+search:
```
Comment (Rugvip) [inline .github/labeler.yml:8]: got `search-common` in packages too

**Review by Rugvip — APPROVED** [review]
Nice! 👍

**Pull request merged**